<launch>
  
  <arg name="robot" default="localhost" doc="allows the user to specify the hostname of the turtlebot"/>

  <arg name="circle" default="true" doc="allows the user to specify the mode of the robot"/>
  <rosparam file="$(find nuturtle_description)/config/diff_params.yaml" />  

  <include file="$(find nuturtle_robot)/launch/basic_remote.launch" >
   <arg name="robot" value="$(arg robot)" />
  </include>

  <!-- <arg name="robot" default="localhost" doc="allows the user to specify the hostname of the turtlebot"/> -->

  <group if='$(arg circle)'>
    <node pkg="nuturtle_robot" type="follow_circle" name="fc">
      <param name="radius" value=".25"/>
      <param name="speed" value=".16"/>  
    </node>
  </group>

  <node pkg="turtlebot3_teleop" type="turtlebot3_teleop_key" name="teleop" if="$(eval arg('circle') ==false)"/>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find nuturtle_description)/config/model.rviz -f odom_frame"/>
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="1000.0" />
    <param name="tf_prefix" value=""/>
  </node>

  <!-- <machine name="turtlebot" address="$(arg robot)" env-loader="/home/ubuntu/loader.sh" user="ubuntu"/> -->
<!-- 
  <node pkg="rosserial_python" type="serial_node.py" name="serial" machine="turtlebot">
    <param name="port" value="/dev/ttyACM0"/>
    <param name="baud" value="115200"/>    
  </node> -->

  <arg name="urdf_file" default="$(find xacro)/xacro '$(find nuturtle_description)/urdf/turtlebot3_burger.urdf.xacro'"/>
  <param name="robot_description" command="$(arg urdf_file)"/>
  <param name="frequency" value = "100"/>

   <node pkg="rigid2d" name="ody" type="odometer" output="screen" machine="turtlebot">
    <param name="odom_frame_id" value ="odom_frame"/>
    <param name="body_frame_id" value = "base_footprint"/>
    <param name="left_wheel_joint" value = "wheel_left_joint"/>
    <param name="right_wheel_joint" value = "wheel_right_joint"/>
  </node>

  <node pkg="nuturtle_robot" name="turtle" type="turtle_interface" output="screen" machine="turtlebot">
    <param name="wheel_base" value ="0.16"/>
    <param name="wheel_radius" value = "0.033"/>
    <param name="left_wheel_joint" value = "wheel_left_joint"/>
    <param name="right_wheel_joint" value = "wheel_right_joint"/>
  </node>



</launch>